packages/ui-app/src/Params/Param/StringBalance.tsx

```diff
@@ -0,0 +1,45 @@
+// Copyright 2017-2018 @polkadot/ui-app authors & contributors
+// This software may be modified and distributed under the terms
+// of the ISC license. See the LICENSE file for details.
+
+import { Props } from '../types';
+
+import React from 'react';
+
+import Input from '../../Input';
+import Bare from './Bare';
+
+export default class StringBalance extends React.PureComponent<Props> {
+  render () {
+    const { className, isDisabled, isError, label, style, withLabel } = this.props;
+
+    return (
+      <Bare
+        className={className}
+        style={style}
+      >
+        <Input
```

Comment:
It needs a pattern to strip invalid chars at the very least. We can add a pattern prop to the base input and have it attached to the raw HTML input - https://www.w3schools.com/tags/att_input_pattern.asp

Reply:
As we discovered, that approach doesn't work, however I've implemented the `maxLength` as you recommended with success